components/layout/display_list_builder.rs

```diff
@@ -1005,9 +1134,7 @@ impl FragmentDisplayListBuilding for Fragment {
         let mut clip = clip.clone();
         clip.intersect_rect(absolute_bounds);
 
-        let grad = self.convert_gradient(absolute_bounds, gradient, style);
-
-        if let Some(x) = grad {
+        if let Some(x) = self.convert_gradient(absolute_bounds, gradient, style) {
```

Comment:
I believe that'd be preferable. Alternatively, let's at least rename `convert_gradient` to a more appropriate method that doesn't seem to imply that it can handle radial gradients.